arangod/Aql/DistributeExecutor.cpp

```diff
@@ -77,6 +79,35 @@ auto DistributeExecutorInfos::getResponsibleClient(arangodb::velocypack::Slice v
   return shardId;
 }
 
+auto DistributeExecutorInfos::shouldDistributeToAll(arangodb::velocypack::Slice value) const
+    -> bool {
+  if (_satellites.empty()) {
+    // We can only distribute to all on Satellite Collections
+    return false;
+  }
+  auto id = value.get(StaticStrings::IdString);
+  if (!id.isString()) {
+    // We can only distribute to all if we can detect the collection name
+    return false;
+  }
+
+  // NOTE: Copy Paste code, shall be unified
```

Comment:
Should use this:

`transaction::helpers::extractCollectionFromId`

However the above method does not detect if _id is invalid. and just returns the string again.
This behavior would lead to weird situations here.
We shall do a refactoring PR to generalize a "CollectionName or Error" variant.